doc/developer-guides/hld/sriov-hld.rst

```diff
@@ -0,0 +1,303 @@
+SR-IOV Virtualization High-Level Design
+=======================================
+
+SR-IOV(single root input/output virtualization) can isolate PCIe devices
+and make the devices to have good performance, close to bare-metal. The
+SR-IOV consists of two basic units, one is called PF(Physical Function)
+that supports SR-IOV PCIe extended capability and manages entire
+physical device, another is called VF (Virtual Function) and is a
+"lightweight" PCIe function which is meant to be given to a VM as a
+pass-through device.
```

Comment:
Can we also use this with pre-launched User VMs?

Regardless of the answer, I think it would be good to add a note about this. :-)